Modules/Loadable/Markups/MRML/vtkMRMLMarkupsNode.cxx

```diff
@@ -923,6 +923,44 @@ void vtkMRMLMarkupsNode::SetNthControlPointPositionWorldFromArray(
   this->SetNthControlPointPositionFromArray(pointIndex, markupxyz, positionStatus);
 }
 
+//-----------------------------------------------------------
+void vtkMRMLMarkupsNode::TransformOrientationMatrixFromNodeToWorld(
+  const double position_Node[3], const double orientationMatrix_Node[9], double orientationMatrix_World[9])
+{
+  if (!this->GetParentTransformNode())
+    {
+    std::copy_n(orientationMatrix_Node, 9, orientationMatrix_World);
+    }
+  else
+    {
+    // Transform orientation matrix from world
+    vtkNew<vtkGeneralTransform> nodeToWorldTransform;
+    this->GetParentTransformNode()->GetTransformToWorld(nodeToWorldTransform);
+    nodeToWorldTransform->TransformVectorAtPoint(position_Node, &orientationMatrix_Node[0], &orientationMatrix_World[0]);
```

Comment:
usually we store matrices like this in an array:

```txt
m0 m1 m2
m3 m4 m5
m6 m7 m8
```

and

```txt
m0 m1 m2 m3
m4 m5 m6 m7
m8 m9 m10 m11
m12 m13 m14 m15
```

Is this convention followed here, too?

If yes, then the vectors that we need to transform are not `orientationMatrix_Node[i], orientationMatrix_Node[i+1], orientationMatrix_Node[i+2]`, but `orientationMatrix_Node[i], orientationMatrix_Node[i+3], orientationMatrix_Node[i+6`].

Reply:
No, convention is not followed here. See: https://github.com/Slicer/Slicer/blob/39be4408f084b3e5d632b373a4985cd7fa6c661b/Modules/Loadable/Markups/MRML/vtkMRMLMarkupsNode.h#L111-L113

I take it then that we should change it to use the normal matrix convention?